### PR TITLE
Closes #289 compile error

### LINKED ISSUE #289: 2.3.7 compile error
[osm2pgrouting-2.3.7-builderror.txt](https://github.com/pgRouting/osm2pgrouting/files/5907335/osm2pgrouting-2.3.7-builderror.txt)

Hi,

Cannot build 2.3.7 on Fedora (GCC 10.2.1 + libpqxx 7.2.0 and 7.3.1) and RHEL 8 (GCC 8.3.1 + libpqxx 7.2.0 ). Errors are attached.

Not sure this is related, but on Fedora I get:

```
CMake Warning (dev) at /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:273 (message):
  The package name passed to `find_package_handle_standard_args` (PQXX) does
  not match the name of the calling package (LibPQXX).  This can lead to
  problems in calling code that expects `find_package` result variables
  (e.g., `_FOUND`) to follow a certain pattern.
Call Stack (most recent call first):
  cmake/FindLibPQXX.cmake:37 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  CMakeLists.txt:14 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.
``` 

...though this message does not appear on RHEL 8.

Any ideas how to fix?

Regards, Devrim

### PR DESCRIPTION
**Pull request by cvvergara**
Changes proposed on this PR:

- Pump up version to 2.3.8
- Minimum cmake version 3.2
- Removing travis tests
- Updated FindPQXX to get the version
- Adding github actions for compilation:
  - Postgres 9.6 to 13
  - libpqxx 6 & 7
- Minimum Version Postgres 9.6 (supported version at this time)
- Minimum Version libpqxx 6
- C++ minimum standard depends on libpqxx
  - For version libpqxx 6: C++14
  - For version libpqxx 7: C++17

Fixes #289, #284  

@pgRouting/admins

### CODE REVIEW

**Review by dkastl — APPROVED** [review]
Thanks!
... and goodbye Travis!

File: NEWS [hunk 1]
```diff
@@ -1,5 +1,18 @@
 osm2pgRouting 2.3.7
```
Comment (cayetanobv) [inline NEWS:1]: New version will be 2.3.8

**Review by cayetanobv — CHANGES_REQUESTED** [review]

File: NEWS (hunk 1 above)
Comment (cvvergara) [reply NEWS]: @cayetanobv 
ready for review again

**Review by cayetanobv — APPROVED** [review]
LGTM

File: NEWS (hunk 1 above)
Comment (cayetanobv) [reply NEWS]: great @cvvergara !! Thanks.

**Review by krashish8 — APPROVED** [review]

**Pull request merged**